src/metabase/pulse/render/body.clj

```diff
@@ -601,66 +567,98 @@
       (let [row-group          (get grouped-rows group-key)
             selected-row-group (mapv #(vector (ffirst %) (first (second %))) row-group)
-            card-name          (or (series-setting viz-settings group-key :name)
-                                   (series-setting viz-settings group-key :title)
-                                   group-key)
-            card-color         (or (series-setting viz-settings group-key :color)
-                                   (nth colors idx))
+            column-name          (or (series-setting viz-settings group-key :name)
+                                   (series-setting viz-settings group-key :title))
             card-type          (or (series-setting viz-settings group-key :display)
                                    chart-type
                                    (nth default-combo-chart-types idx))
             y-axis-pos         (or (series-setting viz-settings group-key :axis)
                                    (nth (default-y-pos viz-settings) idx))]
-        {:name          card-name
-         :color         card-color
+        {:name          column-name
+         :cardName      card-name
          :type          card-type
          :data          selected-row-group
-         :yAxisPosition y-axis-pos}))))
+         :yAxisPosition y-axis-pos
+         :column        (second x-cols)
+         :breakoutValue group-key}))))
+
+(defn- render-multiple-lab-chart
+  "When multiple non-scalar cards are combined, render them as a line, area, or bar chart"
+  [render-type card dashcard {:keys [viz-settings]
+                              :as   data}]
+  (let [viz-settings      (merge viz-settings (:visualization_settings dashcard))
+        multi-res         (pu/execute-multi-card card dashcard)
+        ;; multi-res gets the other results from the set of multis.
+        ;; we shove cards and data here all together below for uniformity's sake
+        viz-settings      (set-default-stacked viz-settings card)
+        viz-settings-seqs (cons viz-settings (map (comp :visualization_settings :card) multi-res))
+        cards             (cons card (map :card multi-res))
+        multi-data        (cons data (map #(get-in % [:result :data]) multi-res))
+        axis-row-fns      (fn [card data]
+                            [(or (ui-logic/mult-x-axis-rowfn card data) #(vector (first %)))
+                             (or (ui-logic/mult-y-axis-rowfn card data) #(vector (second %)))])
+        rowfns            (mapv axis-row-fns cards multi-data)
+        row-seqs          (for [[row-seq [x-rowfn y-rowfn]] (map vector (map :rows multi-data) rowfns)]
+                            (map (comp flatten (juxt x-rowfn y-rowfn))
+                                 (common/row-preprocess x-rowfn y-rowfn row-seq)))
+        col-seqs          (map :cols multi-data)
+        first-rowfns      (first rowfns)
+        [[x-col] [y-col]] ((juxt (first first-rowfns) (second first-rowfns)) (first col-seqs))
```

Comment:
I was tempted to use `(ffirst first-rowfns)` but having to use `(first (second first-rowfns))` seems out of place, so I destructure `x-col` and `y-col` instead since `ui-logic/mult-y-axis-rowfn` returns a list—or a vector I'm not sure—instead of the value itself.

Reply:
there's an argument to omit the `juxt` and just define `x-col` and `y-col` on separate lines, but I've decided not to worry about it